src/scroll-timeline-base.js

```diff
@@ -369,20 +369,24 @@ function getScrollParent(node) {
   if (!node)
     return undefined;
 
-  while (true) {
-    const containingBlock = getContainingBlock(node);
-    if (!containingBlock)
-      return document.scrollingElement;
-
-    const style = getComputedStyle(containingBlock);
+  while (node = getContainingBlock(node)) {
+    const style = getComputedStyle(node);
     switch(style['overflow-x']) {
       case 'auto':
       case 'scroll':
       case 'hidden':
-        return containingBlock;
+        // https://drafts.csswg.org/css-overflow-3/#overflow-propagation
+        // The UA must apply the overflow from the root element to the viewport;
+        // however, if the overflow is visible in both axis, then the overflow
+        // of the first visible child body is applied instead.
+        if (node.tagName != "BODY" ||
+            getComputedStyle(document.scrollingElement).overflow != "visible")
+          return node;
```

Comment:
nit: I think it would be clearer if you flipped the condition and used this case for the exception (the body element in these special circumstances) to return document.scrollingElement and then left the return below as returning the node.

Reply:
done